### PR TITLE
Adding the `gflags` and `rocksdb` packages.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Adding the `gflags` and `rocksdb` packages to make them widely available to CBL-Mariner users.

###### Change Log  

- Added the `rocksdb` package.
- Added the `gflags` package.

###### Does this affect the toolchain?  

No.

###### Test Methodology

- Local package build.

### CODE REVIEW

File: SPECS/gflags/gflags.spec [hunk 1]
```diff
@@ -0,0 +1,64 @@
+Name:           gflags
+Summary:        The gflags package contains a C++ library that implements commandline flags processing. 
+Version:        2.2.2
+Release:        3%{?dist}
+License:        BSD
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://gflags.github.io/gflags/
+Source0:        https://github.com/gflags/gflags/archive/v%{version}.tar.gz
```
Comment (thcrain-msft) [inline SPECS/gflags/gflags.spec:9]: Hi Pawel- is there a reason that Source0 has a non-standard archive name? I know this particular file downloads as `gflags-2.2.2.tar.gz`, and we usually comment out the download URL and set Source0 to %{name}-%{version}.tar.gz.

File: SPECS/rocksdb/rocksdb.spec [hunk 2]
```diff
@@ -0,0 +1,66 @@
+%define _build_id_links none
+
+Name:           rocksdb
+Summary:        A library that provides an embeddable, persistent key-value store for fast storage.
+Version:        6.7.3
+Release:        2%{?dist}
+License:        GPLv2+ and ASL 2.0 and BSD
+URL:            https://rocksdb.org
+Source0:        https://github.com/facebook/rocksdb/archive/v%{version}.tar.gz
```
Comment (thcrain-msft) [inline SPECS/rocksdb/rocksdb.spec:9]: Same comments as for gflags.spec

File: SPECS/gflags/gflags.spec (hunk 1 above)
Comment (thcrain-msft) [inline SPECS/gflags/gflags.spec:9]: Nit: Instances of `gflags` in the Source0 line could be replaced with `%{name}`, to be like our other specs.

Comment (PawelWMS) [reply SPECS/gflags/gflags.spec:9]: Oh snap, thank you! That's a great catch. I've missed the fact that the files are downloaded under a different name.

**Review by thcrain-msft — APPROVED** [review]

**Pull request merged**